### PR TITLE
editorconfig-py: Add recipe

### PR DESCRIPTION
**Pull request by JakobDev**
Another Python lib that is needed by one of my programs.

### CODE REVIEW

File: dev-python/editorconfig-py/editorconfig_py-0.12.3.recipe [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+between different projects. The EditorConfig project maintains \
+a file format and plugins for various text editors which allow this \
+file format to be read and used by those editors. For information \
+on the file format and supported text editors, see the EditorConfig \
+website."
+HOMEPAGE="https://editorconfig.org/"
+COPYRIGHT="2011-2018 EditorConfig Team"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://files.pythonhosted.org/packages/13/31/8382c65fb54ef19b8335f8e05e9ed9f62e33122f9619d989a151cc7e5152/EditorConfig-$portVersion.tar.gz"
```
Comment (korli) [inline dev-python/editorconfig-py/editorconfig_py-0.12.3.recipe:13]: You can define pyName="EditorConfig":
```suggestion
SOURCE_URI="https://pypi.io/packages/source/${pyName:0:1}/$pyName/$pyName-$portVersion.tar.gz"
```

File: dev-python/editorconfig-py/editorconfig_py-0.12.3.recipe [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+CHECKSUM_SHA256="57f8ce78afcba15c8b18d46b5170848c88d56fd38f05c2ec60dbbfcb8996e89e"
+SOURCE_DIR="EditorConfig-$portVersion"
```
Comment (korli) [inline dev-python/editorconfig-py/editorconfig_py-0.12.3.recipe:15]:
```suggestion
SOURCE_DIR="$pyName-$portVersion"
```

**Comment by JakobDev** [comment]
@korli Changes are done now

**Comment by Begasus** [comment]
rename "editorconfig-py"/"editorconfig_py" to "editorconfig" see Gentoo packages (the layout we use): http://gpo.zugaina.org/dev-python/editorconfig

**Comment by JakobDev** [comment]
@Begasus If take a look at the [editorconfig GitHub Group](https://github.com/editorconfig/) you will see that there are libs for multiple languages and plugins for editors. I think it's the best to keep the py to clarify this is the python lib, especially if we may want to pack the other libs in the future.

**Comment by Begasus** [comment]
> If take a look at the [editorconfig GitHub Group](https://github.com/editorconfig/) you will see that there are libs for multiple languages and plugins for editors.

Granted, but it's confusing others it seems also (looking at Repology): https://repology.org/projects/?search=editorconfig
This one is the accurate one imho: https://repology.org/project/python:editorconfig/versions

That also mentions "editorconfig", so far it has been a policy at haikuports that we stick to the Gentoo package layout where possible

**Comment by JakobDev** [comment]
I just thought editoconfig-üy would be a little bit shorter but I changed the to editorcofig-core.py now.

File: dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	cmake
```
Comment (Begasus) [inline dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe:29]: Should be:
```diff
- 	cmake
+ 	cmd:cmake
```

**Comment by Begasus** [comment]
There is a "editorconfig" script installed in $prefix/bin, I added in PROVIDES, but once ran it from the Terminal it needed the python39 packages for this also, I gather the script is linked with the python versions used, so maybe add this to the "PROVIDES-${pythonPackage}"?
Probably would be a good idea to rename it to editorconfig${pythonVersion%.*} see something like this in numpy recipe:
https://github.com/haikuports/haikuports/blob/aff2c5aa26e66c6e280d980e86bf5ae503e1ae86/dev-python/numpy/numpy-1.20.0.recipe#L53

**Comment by JakobDev** [comment]
Why should I add a Provide for every Python Versions if it's only create a file for Python3.9?

**Comment by Begasus** [comment]
> Why should I add a Provide for every Python Versions if it's only create a file for Python3.9?

It creates the binary for every python version, only on the last run (python 3.9) it will be installed (and thus linked to python 3.9)
Not familiar on how the binary is used, but if you have the base package installed and lets say the python 3.8 version installed it will give an error when you try to run it in the Terminal

**Comment by JakobDev** [comment]
@Begasus I hope thsi is waht you mean

**Comment by Begasus** [comment]
Sorry, I'm not trying to make it hard on you, trying to understand how things work on this, or how you plan to use it in the future, will leave a comment in a bit here, thanks for your patience

File: dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	"
+
+PYTHON_PACKAGES=(python3 python38 python39)
+PYTHON_VERSIONS=(3.7 3.8 3.9)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\n\
+	cmd:editorconfig${pythonVersion%.*}\n\
```
Comment (Begasus) [inline dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe:39]: My bad here, like this all cmd's are like editorconfig3 where they should be editorconfig$pythonVersion:
```diff
- 	cmd:editorconfig${pythonVersion%.*}\n\
+	cmd:editorconfig${pythonVersion}\n\
```

File: dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		python=python$pythonVersion
+		installLocation=$prefix/lib/$python/vendor-packages/
+		export PYTHONPATH=$installLocation:$PYTHONPATH
+		mkdir -p $installLocation
+		rm -rf build
+		$python setup.py build install \
+			--root=/ --prefix=$prefix
```
Comment (Begasus) [inline dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe:63]: Add this so it builds the correct pythonVersioned name:
```diff
+		mv $prefix/bin/editorconfig $prefix/bin/editorconfig${pythonVersion}
```

File: dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+		packageEntries  $pythonPackage \
+			$prefix/lib/python*
```
Comment (Begasus) [inline dev-python/editorconfig-core-py/editorconfig_core_py-0.12.3.recipe:66]: Move the binary also to the subpackages:
```diff
- 			$prefix/lib/python*
+ 			$prefix/lib/python* \
+			$prefix/bin
```

**Review by Begasus — COMMENTED** [review]
These should fix the build and ensure every subpackage for the python version has a editorconfig$pythonVersion available
Did you run the tests with haikuporter, they fail here with the change directory

**Comment by JakobDev** [comment]
@Begasus Requested changes are done now

**Comment by Begasus** [comment]
Thanks for your patience, I didn't check a new build (as most changes are requested by myself) :)
Did you check that https://github.com/haikuports/haikuports/pull/5688 build ok with these changes on 64-bit and 32-bit?

**Comment by JakobDev** [comment]
@Begasus This recipe builds with 64bit. Not tested on 32bit, but it's only contains Python code so there should be no difference.

**Comment by Begasus** [comment]
Merged now, keep track on the buildbot if it builds ok on bother 32-bit and 64-bit: see https://github.com/haikuports/haikuports/issues/5668

**Comment by JakobDev** [comment]
Build on 64bit and 32 bit.

**Pull request merged**